runtime/compiler/il/J9MethodSymbol.cpp

```diff
@@ -161,6 +161,7 @@ static TR::RecognizedMethod canSkipNullChecks[] =
    TR::java_util_ArrayList_add,
    TR::java_util_ArrayList_ensureCapacity,
    TR::java_util_ArrayList_get,
+#if JAVA_SPEC_VERSION < 17
    TR::java_lang_String_trim,
```

Comment:
Do all these methods need to be excluded, or can we live we just a subset?

Reply:
It's possible that only a subset needs to be excluded, It requires checking the RI implementation for each method and confirming that it's safe without adding implicit null-checks. Given that the issue is a blocker and as a SOC it would take me sometime, especially with other P high-priority items creeping up.